### PR TITLE
feature/DATA-2377-upgrade-dbt-version-to-1.0.3

### PR DESCRIPTION
**Pull request by oleg-kolesnikov**
## What is this? 
Upgrade using libs:
* dbt-core => 1.0.3
* dbt-postgres => 1.0.3
* dbt-snowflake => 1.0.0

## What changes in this PR? 
* All changes done are respecting updates done in dbt-core v.1.0.3

## How does this impact our users?
* Updates of dbt version for hu-dw-datalake and hu-dw-datawarehouse repos are needed.

## PR Quality
- [v] does `docker-compose exec testxdb test` run successfully?
- [v] does `docker-compose exec testxdb docs` build docs without errors?
- [v] does `docker-compose exec testxdb coverage` pass coverage standards?
- [v] did you leave the codebase better than you found it? 

@Health-Union/hu-data make sure to include a version tag in the merge commit!

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,5 @@
-dbt==0.21.0
+dbt-core==1.0.3
+dbt-postgres==1.0.3
+dbt-snowflake==1.0.0
```
Comment (eculicny) [inline requirements.txt:3]: That's a bit weird that the versions aren't staying synchronized any more, but probably makes some sense at the patch level.

File: dbt_project.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: 'xdb'
 config-version: 2
 version: '0.0.1'
 macro-paths: ["macros"]
-clean-targets: ["target","dbt_modules"]
-require-dbt-version: ">=0.14.0"
+clean-targets: ["target","dbt_packages"]
+require-dbt-version: ">=1.0.0"
```
Comment (eculicny) [inline dbt_project.yml:6]: Do we use xdb in any of the old repos that are using older dbt versions? This change will force anything using xdb to be at least 1.0, which I think is fine. Just need to verify that it doesn't break anything for us.

**Review by eculicny — APPROVED** [review]

File: dbt_project.yml (hunk 2 above)
Comment (Aesthet) [reply dbt_project.yml:6]: I guess only you can know what repositories might use xdb

**Review by Aesthet — APPROVED** [review]

File: dbt_project.yml (hunk 2 above)
Comment (eculicny) [reply dbt_project.yml:6]: True. I don't think it will. The deprecated data models repo doesn't seem to use it at all, so we should be fine on that front.

**Review by eculicny — APPROVED** [review]

**Review by philribbens-hu — APPROVED** [review]

**Pull request merged**